spec/lib/miq_preloader_spec.rb

```diff
@@ -55,45 +56,67 @@
       vms = Vm.where(:ems_id => ems.id)
       vms2 = vms.order(:id).to_a # preloaded vms to be used in tests
 
-      # there is a query for every ems
-      expect { preload(ems, :vms, vms) }.to make_database_queries(:count => 1)
+      # there is now two queries - rails 7, why?
+      expect { preload(ems, :vms, vms) }.to make_database_queries(:count => 2)
+
       expect { preload(ems, :vms, vms) }.not_to make_database_queries
       expect { expect(ems.vms).to match_array(vms2) }.not_to make_database_queries
       # vms does not get loaded, so it is not preloaded
-      expect { expect(vms.first.ext_management_system).to eq(ems) }.to make_database_queries(:count => 2)
+      expect { expect(vms.first.ext_management_system).to eq(ems) }.to make_database_queries(:count => 1)
     end
 
     it "preloads with a loaded relation (records is a relation)" do
       ems
       emses = ExtManagementSystem.all.load
       vms   = Vm.where(:ems_id => emses.select(:id)).load
-      expect { preload(emses, :vms, vms) }.not_to make_database_queries
+
+      # there is now two queries - rails 7, why?
+      expect { preload(emses, :vms, vms) }.to make_database_queries(:count => 1)
```

Comment:
ok, first thought is that this should be 0.

I did make changes last night in virtual attribute that specifically addresses this use case - maybe it is back down to 0?

If not, lets see if we can add this use case to the `virtual_includes` specs

Reply:
feel free to push any tests you think we're missing here